🚀 2단계: 요구 사항 정리

dev-yoonjung opened the pull request:
안녕하세요, 리뷰어님. 

2단계 미션 완료하여 PR 올립니다. 
추가로 1단계 미션에 대해 몇가지 수정하여 같이 올립니다. 

잘 부탁드립니다. 😊

### 요구 사항
- [x] kitchenpos 패키지의 코드를 보고 키친포스의 요구 사항을 README.md에 작성한다.

dev-yoonjung committed:
refactor(car): Rename racing car package

dev-yoonjung committed:
feat(positive): Implement add value

dev-yoonjung committed:
feature(positive): Override equals and hash code

dev-yoonjung committed:
docs: Define requirements for menu group

dev-yoonjung committed:
docs: Define requirements for menu

dev-yoonjung committed:
docs: Define requirements for table

dev-yoonjung committed:
docs: Define requirements for order

dev-yoonjung committed:
docs: Define requirements for product

dev-yoonjung committed:
docs: Update requirement for product

kyucumber commented on README.md:
```diff
@@ -9,6 +9,72 @@ docker compose -p kitchenpos up -d
+    - [ ] 메뉴 공개 여부에 대해서 반드시 입력해야 한다.
+  - [ ] 모든 메뉴 목록을 보여준다.
+  - [ ] 메뉴를 숨길 수 있다.
+  - [ ] 메뉴를 공개할 수 있다.
```
메뉴를 공개할 때 아래와 같은 조건이 있었던 것 같아요. 이 부분도 메뉴를 공개할 수 있다 하위에 요구사항으로 포함시키면 어떨까요?

```
if (menu.getPrice().compareTo(sum) > 0) {
```

kyucumber commented on README.md:
```diff
@@ -9,6 +9,72 @@ docker compose -p kitchenpos up -d
+  - [ ] 모든 주문 목록을 보여준다.
+- 제품
+  - [ ] 제품을 등록할 수 있다.
+    - [ ] 제품명은 반드시 입력해야 한다.
```
제품명에서도 아래와 같이 비속어 체크를 하고 있어요. 요구사항에 추가하면 어떨까요?

```
purgomalumClient.containsProfanity(name)
```

kyucumber commented on README.md:
```diff
@@ -9,6 +9,72 @@ docker compose -p kitchenpos up -d
+    - [ ] 배달, 테이크아웃, 매장식사 중 고를 수 있다.
+    - [ ] 배달 주문의 경우 반드시 배달 주소를 작성해야 한다. 
+    - [ ] 매장식사 주문의 경우 반드시 유효한 테이블이어야 하며, 손님이 앉아 있어야 한다.
+    - [ ] 유효한 메뉴를 반드시 0개 이상 선택해야 한다.
```
유효한 메뉴의 개수 뿐만 아니라 수량(quantity)에 대한 요구사항도 있었던 것 같은데요. 관련해서도 추가되면 좋을 것 같네요.

kyucumber commented on README.md:
```diff
@@ -9,6 +9,72 @@ docker compose -p kitchenpos up -d
+    - [ ] 이미 사람이 앉아 있는 유효한 테이블에 한해 0명 이상으로 변경할 수 있다.
+- 주문
+  - [ ] 주문을 등록할 수 있다.
+    - [ ] 배달, 테이크아웃, 매장식사 중 고를 수 있다.
```
각 주문 항목 별 요구사항을 상세하게 잘 정리해주셨네요. 👍 

현재 주문 순서가 한눈에 보이지는 않는데, 요구사항 명세를 통해 도메인을 이해하는데 도움이 되는 경우도 있으므로 아래와 같이 주문 타입 별 상태가 어떻게 흘러가는지에 대해 정리해보아도 좋을 것 같네요.

- 배달 주문
  - WAITING -> ACCEPTED -> DELIVERING -> ...

kyucumber reviewed (changes requested):
요구사항 정리를 꼼꼼하게 잘 해주셨네요.

간단한 몇가지 코멘트 남겨두었는데요, 확인 후 다시 요청해주시면 머지하도록 하겠습니다. 🙇

dev-yoonjung committed:
docs: Update description for kitchen-pos

- 메뉴 공개 수정에 대한 요구사항 추가
- 제품명 유효성 검사에 대한 요구사항 추가
- 주문할 때의 메뉴 수량에 대한 요구사항 추가
- 주문 흐름에 따른 상태관리 요구사항 추가

kyucumber reviewed (approved):
요구사항 잘 정리해주셨습니다.

다음 단계 진행을 위해 머지할게요.
